### PR TITLE
[enhancement] new memory handling mechanism for EwoK

### PR DESCRIPTION
**Pull request by PThierry**
#### Description:

- Differentiated memory handling for mass storage and RAM consumption for each task
- Separation between portable informations and SoC specific informations into to separated Ada file
- Handling of memory-model specific actions in corresponding package (typically ewok-mpu), through abstracted calls done by memory abstraction layer ewok.memory
- Cleaner application data recopy and bss zeroification
- strict application and kernel mapping into their own memory space. Any overload fail at link time
- memory layout is defined through simple dictionary named 'socinfo.cfg', used to generate the kernel headers, reducing the amount of needed code in the kernel
- all generated informations are hold in the config package:
   - config.applications: portable information, about application ELF size and constraints
   - config.memlayout: SoC specific information, about the way the memory layout is handling for applications and kernel
- Applications do not need to declare NUMSLOTS in their configuration. They only have to declare STACKSIZE and HEAPSIZE. Other informations are calculated
- Memory overload is detected and pretty printed at header generation time, before compilation

#### Type:

- IMPROVEMENT

#### Rationale:

- Simplify the way the memory layout is handled
- Differentiated layouting between mass storage and RAM permits to get back some free memory in some cases (RAM or flash)
- No region, slot or subregion notion exists at application or kernel high level layer. The kernel memory handling is ready to support various MPU systems and MMU through memory abstraction package

### CODE REVIEW

**Comment by PThierry** [comment]
**IMPORTANT**: This PR depends on tataouine and libstd corresponding PR to build properly

**Comment by PThierry** [comment]
global inter-PR dependencies. Exhaustively checked manually in replacement of CI

File: tools/devmap/gen_app_final_ld.pl [hunk 1]
```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/perl -l
+#
+
+use strict;
+# getting back script path
+use File::Basename qw(dirname basename);
+use File::Path qw(make_path);
+use File::Path qw(make_path);
```
Comment (ptrebuc) [inline tools/devmap/gen_app_final_ld.pl:8]: duplicate previous line

File: tools/devmap/lib/Ada/Format.pm [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    return $appname;
+}
+
+#
+# Formatting:
+#
+# save generic application information into cfg file $FH
+#
+sub format_appinfo_for_cfg {
+    local *FH = shift;
```
Comment (ptrebuc) [inline tools/devmap/lib/Ada/Format.pm:122]: could use a scalar here (since Perl 5.6 :-) )

**Pull request merged**